src/log-user-session.c

```diff
@@ -495,8 +505,24 @@ void write_log(int fd_input, int fd_log, char *buffer, size_t size) {
     }
 }
 
+int is_command_whitelisted(const char *original_command) {
+    // retrieve first token which is the command
+    char *command = strtok(strdup(original_command), " \n\t");
+
+    int i;
+    for (i = 0; i < *whitelist_size ; i++) {
+        if ( !strcmp(command, opt_command_whitelist[i]) ) {
+            free(command);
```

Comment:
I don't understand why you have to free command here

Reply:
alright.. strdup allocates mem.. and should be freed.

Reply:
There is a memory leak if the command is not whitelisted